src/components/DataTableInsertRow.vue

```diff
@@ -0,0 +1,117 @@
+<template>
+  <tr class="data-table-insert-row">
+    <td class="key-column" @click="startCreating('keyInput')">
+      <input
+        v-if="isCreating"
+        ref="keyInput"
+        type="text"
+        v-model="key"
+        @keydown.enter="stopCreating"
+        @keydown.esc="stopCreating(false)"
+        @blur="handleBlur"
```

Comment:
seems like if you are currently focused on the key field, enter a value, press tab, and then enter a value, then press enter, the key-value pair is not saved.

Reply:
@emanlodovice Can you please try this again with the latest commit? Note that adding a new record using a key that already exists counts as just editing that existing key.